app/main/download_data.py

```diff
@@ -238,3 +240,16 @@ def process_api_response(query_params: dict) -> tuple:
         return abort(500), f"Unknown content type: {content_type}"
 
     return content_type, file_content
+
+
+def process_async_download(query_params: dict):
+    """Calls data-store for a file download request.
+
+    :param query_params: Query parameters for the API request.
+    """
+    request_url = (
+        Config.DATA_STORE_API_HOST
+        + "/trigger_async_download"
+        + ("?" + urlencode(query_params, doseq=True) if query_params else "")
+    )
+    requests.post(request_url)
```

Comment:
This should come back with a 204 - I think we should check the response here and raise an error if appropriate; consider using [raise_for_status()](https://3.python-requests.org/api/?highlight=raise_for_status#requests.Response.raise_for_status) at least. This is because, if the API requests returns a 500, we probably should show an error page to the user as well.

Reply:
Added a check for status code to app/main/routes.py/download(), starting at line 21, but unsure if that's what you want here.